common/types/endpoint.go

```diff
@@ -343,3 +414,38 @@ func (e *Endpoint) InvalidatePolicy() {
 		e.Consumable.Iteration = 0
 	}
 }
+
+func (e *Endpoint) LogStatus(code StatusCode, msg string) {
```

Comment:
So this creates a stack of log messages which are categorized which is great. I don't think we can always refer to the last entry as the current status though. A warning may be followed by an OK log message but the user will still want to see the warning. A reconfiguration event by the user may fix the warning in which case the status should then switch back to OK.

I think we'll need both, a log of relevant status messages which preserve the history for particular endpoint transitions.

In addition to that, a managed status for health field which represents the current state of the endpoint:
 - OK: Endpoint is functional in the configuration provided
 - Build: Endpoint is currently undergoing a rebuild of the programs
 - Warn: A non fatal warning has occurred since the last configuration/poilcy change but the endpoint should be partially operational
 - Fail: A fatal warning has occurred, the endpoint is not functional.